From e7a09a6262abe6181514a0b64f79acec8bc3c405 Mon Sep 17 00:00:00 2001
From: Maxime Ripard <maxime@cerno.tech>
Date: Wed, 23 Jun 2021 11:56:56 +0200
Subject: [PATCH] ARM: dts: rpi: Add the firmware node to vc4

Add the firmware phandle to the vc4 node so that we can send it the
message that we're done with the firmware display.

Signed-off-by: Maxime Ripard <maxime@cerno.tech>
---
 arch/arm/boot/dts/bcm2711-rpi.dtsi | 4 ++++
 arch/arm/boot/dts/bcm2835-rpi.dtsi | 4 ++++
 2 files changed, 8 insertions(+)

--- a/arch/arm/boot/dts/bcm2711-rpi.dtsi
+++ b/arch/arm/boot/dts/bcm2711-rpi.dtsi
@@ -35,6 +35,10 @@
 	};
 };
 
+&vc4 {
+	raspberrypi,firmware = <&firmware>;
+};
+
 &cma {
 	/* Limit cma to the lower 768MB to allow room for HIGHMEM on 32-bit */
 	alloc-ranges = <0x0 0x00000000 0x30000000>;
--- a/arch/arm/boot/dts/bcm2835-rpi.dtsi
+++ b/arch/arm/boot/dts/bcm2835-rpi.dtsi
@@ -76,6 +76,10 @@
 	power-domains = <&power RPI_POWER_DOMAIN_USB>;
 };
 
+&vc4 {
+	raspberrypi,firmware = <&firmware>;
+};
+
 &vec {
 	power-domains = <&power RPI_POWER_DOMAIN_VEC>;
 	status = "okay";
